packages/widget_toolkit_qr/lib/src/lib_qr_scanner/views/qr_scanner_component.dart

```diff
@@ -141,28 +141,26 @@ class QrScannerComponent<T> extends StatelessWidget {
           ),
           if (isLoadingIndicatorVisible == null) ...[
             SizedBox(
-              height: spaceBetweenScannerAndLoadingWidget ?? 20,
+              height: spaceBetweenScannerAndLoadingWidget ??
+                  context.widgetToolkitTheme.spacingL,
             ),
-            _buildLoadingLinearProgressIndicator(context),
+            RxBlocBuilder<QrScannerBlocType<T>, bool>(
+              state: (bloc) => bloc.states.isLoading,
+              builder: (ctx, state, bloc) => state.hasData && state.data == true
```

Comment:
@PVGeorgievPrime, I guess there was a misunderstanding but I was promoting to keep the widgets as part of the `build` method not the closures such as `builder: (ctx, state, bloc)`. It's perfectly ok to have a private method for each closure so that your `build` method remains easy to read

Reply:
reworked